rts/motoko-rts/src/memory.rs

```diff
@@ -0,0 +1,57 @@
+#[cfg(feature = "ic")]
+pub mod ic;
+
+use crate::rts_trap_with;
+use crate::types::*;
+
+use motoko_rts_macros::ic_mem_fn;
+
+/// A trait for heap allocation. RTS functions allocate in heap via this trait.
+///
+/// To be able to link the RTS with moc-generated code, we implement wrappers around allocating
+/// functions that pass `ic::IcMemory` for the `Memory` arguments, and export these functions with
+/// the expected names for the generated code. For example, for a function like
+///
+/// ```
+/// unsafe fn allocating_function<M: Memory>(mem: &mut M) { ... }
+/// ```
+///
+/// we implement (or generate with a macro)
+///
+/// ```
+/// #[no_mangle]
+/// unsafe extern "C" fn export_name() { allocating_function(crate::memory::ic::IcMemory) }
```

Comment:
Is `export_name` just `allocating_function`?

Reply:
`export_name` is whatever we want to call this function in the .wasm file generated for the RTS.  Without `no_mangle` the compiler mangles the name, so for example `allocating_function` above will have a name like `motoko_rts::allocation_function::h19cee68178dfb200`. We add `no_mangle` to the wrapper function and then give it a name that we later use to refer to from the generated code.

Does this make sense? Perhaps I should update this comment to make it more clear.